### PR TITLE
Remove signing for mac dmgs, as it is broken

### PR DESCRIPTION
**Pull request by poshul**
## Description
Fix https://github.com/OpenMS/OpenMS/issues/6411
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by poshul** [comment]
Fixes https://github.com/OpenMS/OpenMS/issues/6411

File: cmake/package_dragndrop_dmg.cmake [hunk 1]
```diff
@@ -92,29 +88,6 @@ add_custom_target(dist
...
-########################################################### Create dmg with background image
-if (DEFINED CMAKE_VERSION AND NOT "${CMAKE_VERSION}" VERSION_LESS "3.5")
-  set(OPENMS_LOGO ${PROJECT_SOURCE_DIR}/cmake/MacOSX/openms_logo_large_transparent.png) ## The logo to be used for the OpenMS folder on the DMG
-  set(OPENMS_DMG_FOLDER_NAME "${CPACK_PACKAGE_NAME}-${OPENMS_PACKAGE_VERSION_FULLSTRING}") ## The name of the OpenMS folder on the DMG
-  configure_file(${PROJECT_SOURCE_DIR}/cmake/MacOSX/setup_applescript.scpt.in ${PROJECT_BINARY_DIR}/macOS_bundle_setup/setup_applescript.scpt)
-  set(CPACK_DMG_DS_STORE_SETUP_SCRIPT ${PROJECT_BINARY_DIR}/macOS_bundle_setup/setup_applescript.scpt)
-  #Next line could overcome a script but since we do not have a fixed name of the OpenMS-$VERSION folder, it probably won't work
-  #set(CPACK_DMG_DS_STORE ${PROJECT_SOURCE_DIR}/cmake/MacOSX/DS_store_new)
-  set(CPACK_DMG_BACKGROUND_IMAGE ${PROJECT_SOURCE_DIR}/cmake/MacOSX/background.png)
-  set(CPACK_DMG_FORMAT UDBZ) ## Try bzip2 to get slighlty smaller images
```
Comment (jpfeuffer) [inline cmake/package_dragndrop_dmg.cmake:104]: This would be really nice to keep.

Comment (poshul) [reply cmake/package_dragndrop_dmg.cmake:104]: My takeaway from our discussion yesterday was that DMG signing was not feasible at present, and Apple was likely to only make it harder in the future. I can add a comment leaving breadcrumbs to this issue if you think that would help.

Comment (jpfeuffer) [reply cmake/package_dragndrop_dmg.cmake:104]: But this specific part is for designing the DMG style. Otherwise it is really ugly.

Comment (jpfeuffer) [reply cmake/package_dragndrop_dmg.cmake:104]: We have to use DMG until someone implements PKG

Comment (poshul) [reply cmake/package_dragndrop_dmg.cmake]: Okay. Does this look better now?

**Pull request merged**